src/tests/resistance/fuzzing_flood.rs

```diff
@@ -0,0 +1,640 @@
+struct Stats {
+    handshake_accepted: u16,
+    current_connections: u16,
+    max_active_connections: u16,
+    handshake_latencies: Vec<Duration>,
+    handshake_rejected: u16,
+    peers_dropped: u16,
+    reply_latencies: Vec<Duration>,
+    corrupt_terminated: u16,
+    corrupt_rejected: u16,
+    reply_instead_of_termination: u16,
+    corrupt_ignored: u16,
+
+    reply_errors: u16,
+    read_errors: u16,
+    write_errors: u16,
+    handshake_errors: u16,
+    dangling: u16,
+}
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 8)]
+async fn rising_fuzz() {
+    // ZG-RESISTANCE-006
+    //
+    // Simulate high load situations, while also sending corrupt messages.
+    //
+    // We simulate active peers which send 100 requests (and validate the replies), and finally send a corrupt message
+    // which ends the connection. We measure the latency of the replies and the type of response to the corrupt message.
+    //
+    //  *NOTE* run with `cargo test --release tests::resistance::fuzzing_flood::rising_fuzz -- --nocapture`
+    //
+    // Currently only works for zcashd as it requires block seeding.
+    //
+    // Example zcashd run result (notably some connections appear to hang after sending the corrupt message, and some
+    // get replies to the corrupt message).
+    //
+    // Stats
+    // ╭───────┬──────────┬────────────┬───────────┬───────────┬───────────┬────────────┬────────────┬──────────┬─────────┬─────────┬───────────┬─────────────┬────────────┬──────────╮
+    // │ Peers ┆ Requests ┆ Max active ┆ Handshake ┆ Handshake ┆ Handshake ┆ Connection ┆   Corrupt  ┆  Corrupt ┆ Corrupt ┆ Corrupt ┆ IO Errors ┆ Bad replies ┆    Hung    ┆ Time (s) │
+    // │       ┆          ┆            ┆  Accepted ┆  Rejected ┆   Errors  ┆   Dropped  ┆ Terminated ┆ Rejected ┆ Ignored ┆ Replied ┆           ┆             ┆ Connection ┆          │
+    // ╞═══════╪══════════╪════════════╪═══════════╪═══════════╪═══════════╪════════════╪════════════╪══════════╪═════════╪═════════╪═══════════╪═════════════╪════════════╪══════════╡
+    // │     1 ┆      100 ┆          1 ┆         1 ┆         0 ┆         0 ┆          0 ┆          1 ┆        0 ┆       0 ┆       0 ┆         0 ┆           0 ┆          0 ┆     0.03 │
+    // ├╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┤
+    // │    10 ┆      100 ┆         10 ┆        10 ┆         0 ┆         0 ┆          1 ┆          5 ┆        0 ┆       1 ┆       1 ┆         0 ┆           0 ┆          2 ┆    10.72 │
+    // ├╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┤
+    // │    20 ┆      100 ┆         18 ┆        20 ┆         0 ┆         0 ┆          0 ┆         14 ┆        0 ┆       3 ┆       1 ┆         0 ┆           0 ┆          2 ┆    21.94 │
```

Comment:
What's "max active", I notice it's lower than the peer count/accepted handshakes in some cases? Are these expected count - hung at the peak?

Reply:
'max active' is the peak number of concurrent connections. Maybe `peak` is a better word then :D The table headers in general could use an improvement.